src/index.ts

```diff
@@ -123,15 +137,27 @@ export type APIResult<T extends APIKey, U extends string = T> = {
 };
 
 const VENDOR_TOKEN_LENGTH = 60;
+const DEFAULT_MAX_RETRY = 3;
+const DEFAULT_RETRY_SLEEP_TIME = 500;
 
 export class Cinnamon {
-    private config: Config;
+    private config: AugmentedRequired<Config, 'maxRetry' | 'retrySleepTime'>;
     private refreshToken = '';
     private token = '';
     private refreshTokenRequest: Promise<null> | null = null;
 
-    constructor(config: Config) {
-        this.config = config;
+    constructor({
+        url,
+        retryHook,
+        maxRetry = DEFAULT_MAX_RETRY,
+        retrySleepTime = DEFAULT_RETRY_SLEEP_TIME,
```

Comment:
maybe just spread the non-default params so you don't have to pass them around (in case Config interface changes)

Reply:
fixed